### ISSUE TITLE
Add support for self-hosted Gitlab(CE) or GitHub Enterprise to run CI/CD workflows in Amplify

### ISSUE THREAD

**Issue by cloudgeek7** [aws-amplify/amplify-hosting#2609]
**Is your feature request related to a problem? Please describe.**
We cannot enable CI/CD workflows for self-hosted gitlab repositories in Amplify. It only works for the GitHub and Gitlab public versions. There are many customers I support who are blocked by the lack of this feature. Customers are adopting workarounds or moving away from Amplify.

**Describe the solution you'd like**
Currently the Amplify console has support for GitHub and Gitlab for CI/CD workflows, but it doesn't support self-hosted (enterprise) version.

**Additional context**
As a workaround we are using Gitlab Runners on Amazon EC2 or using S3, CodeCommit and CodePipeline which is not a proper solution.

**Comment by Jay2113** [comment]
Closing this feature request since its a duplicate of #14
